math/primes.go

```diff
@@ -0,0 +1,32 @@
+package math
+
+import (
+	"crypto/rand"
+	"io"
+	"math/big"
+)
+
+// IsSafePrime reports whether p is (probably) a safe prime.
+// The prime p=2*q+1 is safe prime if both p and q are primes.
+// The n parameter is the number of checks for primality test.
```

Comment:
Good to add the caveat from `ProbablyPrime`:

```
ProbablyPrime is not suitable for judging primes that an adversary may
have crafted to fool the test.
````

Reply:
added